### PR TITLE
Fix for sense block digging (now only leaves)

### PR DESCRIPTION
**Pull request by 4gname**

### CODE REVIEW

**Review by draknyte1 — CHANGES_REQUESTED** [review]
I’d check if the block is leaves by ore dictionary, not just vanilla.

**Comment by 4gname** [comment]
Why hasn't anyone done it for so many years?

**Comment by repo-alt** [comment]
I remember somebody fixing this stuff, (or at least talking that he will)

**Comment by draknyte1** [comment]
> Why hasn't anyone done it for so many years?

Probably because nobody uses the item.

**Comment by bartimaeusnek** [comment]
I don't quite get the usefulness of this pr...

**Comment by draknyte1** [comment]
> I don't quite get the usefulness of this pr...

A sense can currently be used to mine anything, Totally not intended.

**Comment by bartimaeusnek** [comment]
> 
> 
> > I don't quite get the usefulness of this pr...
> 
> A sense can currently be used to mine anything, Totally not intended.

It can be used to mine everything that doesnt need any tools, thats correct, i literally changed that a while ago.

**Comment by 4gname** [comment]
> It can be used to mine everything that doesnt need any tools, thats correct, i literally changed that a while ago.

No, you didn't fix this problem

**Comment by Prometheus0000** [comment]
This could be easily checked by going ingame and trying it out if anyone wants to. I thought it was for harvesting crops, but that seems like a worse use than getting rid of leaves anyway.

**Comment by 4gname** [comment]
> I’d check if the block is leaves by ore dictionary, not just vanilla.

I will correct the commit later, add instanceof?

File: src/main/java/gregtech/common/tools/GT_Tool_Sense.java [hunk 1]
```diff
@@ -42,9 +42,8 @@ public int convertBlockDrops(List<ItemStack> aDrops, ItemStack aStack, EntityPla
             for (int i = -2; i < 3; i++) {
                 for (int j = -2; j < 3; j++) {
                     for (int k = -2; k < 3; k++) {
-
-                        if (!((aEvent.world.getBlock(aX + i, aY + j, aZ + k) == Blocks.leaves) ||
-                                (aEvent.world.getBlock(aX + i, aY + j, aZ + k) == Blocks.leaves2))) {
+                        Block gBlock = aEvent.world.getBlock(aX + i, aY + j, aZ + k);
+                        if (!(gBlock instanceof BlockLeaves)) {
```
Comment (4gname) [inline src/main/java/gregtech/common/tools/GT_Tool_Sense.java:46]: I don't know how to check oredict for a block from the world

**Comment by Dream-Master** [comment]
@draknyte1 can this be merged ?

**Comment by bartimaeusnek** [comment]
1. The change need to be applied to ``isMinableBlock`` not ``convertBlockDrops``
2. I don't see the point in restricting this to leaves and plants only.

**Comment by repo-alt** [comment]
Because digging dirt with a scythe makes so much sense...

**Comment by 4gname** [comment]
> 1. The change need to be applied to `isMinableBlock` not `convertBlockDrops`
> 2. I don't see the point in restricting this to leaves and plants only.

isMinableBlock this is only for breaking the block, and cutting out along the radius is already convertBlockDrops

**Comment by bartimaeusnek** [comment]
> 
> 
> Because digging dirt with a scythe makes so much sense...

So does punching trees, or picking up sand/gravel/dirt with your hands.

**Comment by 4gname** [comment]
> > Because digging dirt with a scythe makes so much sense...
> 
> So does punching trees, or picking up sand/gravel/dirt with your hands.

no, you do not dig around the radius with your hands

**Comment by Prometheus0000** [comment]
Does it really make sense for a scythe to dig dirt in a 5x5 area? Or chop wood? Plus we have specialized tools for those things.

**Comment by bartimaeusnek** [comment]
> Does it really make sense for a scythe to dig dirt in a 5x5 area? Or chop wood? Plus we have specialized tools for those things.

yupp, which have a speed boost, that the scythe does not provide.

**Comment by Cleista** [comment]
I like not dismantling my entire base when trying to harvest crops with a Sense. 

- [Try to harvest crops](https://cdn.discordapp.com/attachments/181078474394566657/728317113197920417/unknown.png)
- [Accidentally dismantle base instead](https://cdn.discordapp.com/attachments/181078474394566657/728317188267835401/unknown.png)

This is **not expected to occur** and I doubt people not named bart will genuinely disagree with me on this claim, considering other users have raised issues like [#5956](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/5956) specifically regarding this nonsensical behavior of the Sense tool that prevents it from being used for _the one task it's designed to do_: crop and leaf harvesting.

**Comment by 4gname** [comment]
> I like not dismantling my entire base when trying to harvest crops with a Sense.
> 
> * [Try to harvest crops](https://cdn.discordapp.com/attachments/181078474394566657/728317113197920417/unknown.png)
> * [Accidentally dismantle base instead](https://cdn.discordapp.com/attachments/181078474394566657/728317188267835401/unknown.png)
> 
> This is **not expected to occur** and I doubt people not named bart will genuinely disagree with me on this claim, considering other users have raised issues like [#5956](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/5956) specifically regarding this nonsensical behavior of the Sense tool that prevents it from being used for _the one task it's designed to do_: crop and leaf harvesting.

I explain this by the fact that Bart never played gregtech, but only codes it

**Comment by Dream-Master** [comment]
I reopen it. Feel free to discuss further.

File: src/main/java/gregtech/common/tools/GT_Tool_Sense.java (hunk 1 above)
Comment (repo-alt) [reply src/main/java/gregtech/common/tools/GT_Tool_Sense.java:47]: There is literally Block.isLeaves function

Comment (repo-alt) [reply src/main/java/gregtech/common/tools/GT_Tool_Sense.java:47]: isFoliage is probably also worth checking.

**Comment by bartimaeusnek** [comment]
> > I like not dismantling my entire base when trying to harvest crops with a Sense.
> > 
> > * [Try to harvest crops](https://cdn.discordapp.com/attachments/181078474394566657/728317113197920417/unknown.png)
> > * [Accidentally dismantle base instead](https://cdn.discordapp.com/attachments/181078474394566657/728317188267835401/unknown.png)
> > 
> > This is **not expected to occur** and I doubt people not named bart will genuinely disagree with me on this claim, considering other users have raised issues like [#5956](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/5956) specifically regarding this nonsensical behavior of the Sense tool that prevents it from being used for _the one task it's designed to do_: crop and leaf harvesting.
> 
> I explain this by the fact that Bart never played gregtech, but only codes it

Just having an UV Base on Delta doesn't count, right?

No, it doesn't make SENSE to impair your ability to harvest shit you can w/o any tool.

**Pull request closed without merging**